endpoints/v2/manifest.py

```
@@ -517,6 +519,8 @@ def _write_manifest(
         raise ManifestInvalid(detail={"message": str(cme)})
     except RetargetTagException as rte:
         raise ManifestInvalid(detail={"message": str(rte)})
+    except ImmutableTagException as ite:
+        raise TagImmutable(detail={"message": f"tag '{ite.tag_name}' is immutable"})
```

Ruff (0.14.11)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)